### ISSUE TITLE
Doubt regarding Faster RCNN features

### ISSUE THREAD

**Issue by PranitChawla** [crodriguezo/DORi#3]
Are the faster rcnn features sampled from only one key frame or a set of key frames?

`            auxx = torch.cat((nodeQueryVH.unsqueeze(1).repeat(1,nodeVideo.size(1), 1), updateNodeVideo), dim=2)
            VQ   = self.video_query(auxx)
            auxx = torch.cat((nodeQueryVH.unsqueeze(1).unsqueeze(1).repeat(1,nodeHuman.size(1), nodeHuman.size(2), 1), updateNodeHuman), dim=3)
            HQ   = self.human_query(auxx)
            MessageHQV = torch.cat((VQ, torch.sum(HQ,dim=2)), dim=2)`
In the above code, I was getting an error saying that size of HQ and VQ dont match. VQ has I3D features sampled at 25 fps and HQ will have faster rcnn features sampled only from the keyframes, so how will the dimensions match? Thanks

**Comment by crodriguezo** [comment]
Hi Pranit,

Thank you for your message. Well, I3D generates a feature representation every ~8 frames. We use pascal variance to select keyframes for each I3D feature (activity representation). Thus, we choose one keyframe for every 8 frames to extract FasterRCNN features. In the preprocessing JSON files, you can find the `frame_selected` for each video.

**Comment by PranitChawla** [comment]
Hi, 

Thank you for your reply. I think this will solve the issue for the difference of nodes in the graph. I had one more doubt. If the video has a 1000 frames (0-1000) and lets say the temporal ground truth is from (200-300), does the I3D feature vector have size (1000Xdim)? where fdim is the feature dimension of the vector? Or does it have a size of ((1000/8)Xfdim), since I3D features are sampled every 8th frame.
